File: class23/cpp/AP2301-Exemplo01.cpp
#include <iostream>
#include <iomanip>
#include <cmath>
double potencia(double base, int exp) {
    if (exp == 0) return 1;
    double res = 1;
    for (int i = 0; i < abs(exp); i++) res *= base;
    return exp < 0 ? 1/res : res;
}
int main() {
    double b; int e;
    std::cout << "Base: "; std::cin >> b;
    std::cout << "Expoente: "; std::cin >> e;
    std::cout << b << "^" << e << " = " << std::fixed << std::setprecision(4) << potencia(b, e) << std::endl;
    return 0;
}

File: class23/java/AP2301-Exemplo01.java
import java.util.Scanner;
public class FuncaoPotencia {
    static double potencia(double base, int exp) {
        if (exp == 0) return 1;
        double res = 1;
        for (int i = 0; i < Math.abs(exp); i++) res *= base;
        return exp < 0 ? 1/res : res;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Base: "); double b = sc.nextDouble();
        System.out.print("Expoente: "); int e = sc.nextInt();
        System.out.printf("%.1f^%d = %.4f%n", b, e, potencia(b, e));
        sc.close();
    }
}